### ISSUE TITLE
Extending Unreal classes

### ISSUE THREAD

**Issue by peteshand** [RobertBorghese/Haxe-UnrealEngine5#7]
Hey do you have any advice around extending unreal objects other than `Actor`, eg: `UBlueprintFunctionLibrary`, `ActorComponent` or `SceneComponent`? I'm getting errors along the lines of:

`error C2039: '__SGetClass': is not a member of 'USceneComponent'`

**Comment by SomeRanDev** [comment]
Oh weird... I actually haven't tried to extend them yet, but that error seems to be the result of all UObject Haxe classes using `@:nativeGen`. There's probably something that's triggering Haxe to assume it can use reflection/dynamic features on those classes when it can't. I'll take a look.

**Comment by SomeRanDev** [comment]
Okay, based on my tests, one of the issues are the cpp/h output file names, and the .generated.h name is wrong. They shouldn't have the prefixes. I think you brought this up before, but I never got around to fixing since it was working fine with actors. (;^_^) Components definitely need to meet these conditions. So I just pushed a commit to output correct file names.

However, I also believe component classes need to have "Component" at the end of their name, so I also need a system for automating that from the shortened "Comp". So there's still a bit to test and work on.

**Comment by peteshand** [comment]
Oh.. just reading your last comment now.. so this sounds related to the `A` prefix issue I just raised.. I should have probably just added the info here. oh ok.. so sounds like there is a little bit to it

**Comment by SomeRanDev** [comment]
You're all good XD I have a working component using this code if you're curious.
```haxe
package game;

import ue.*;

@:uclass(ClassGroup=(Custom), meta=(BlueprintSpawnableComponent))
class TestComponent extends SceneComp {
	@:uprop public var bla = 43;

	public function new() {
	}
}
```
There's still a lot of problems. Like I mentioned before, I want to be able to call the class anything, so need to automate the "Component" suffix. Also, since I haven't written any of the manual bindings for USceneComponent, you cannot "override" the BeginPlay, so it crashes for some reason? Maybe it's for another reason? Idk.

The meta comes from the meta that's automatically added when you create a USceneComponent class in the Unreal editor. I'm not really sure what it does, but the component doesn't appear to work without it, so might automate that as well.

Long story short, components aren't complete yet, but should be by tomorrow.

**Comment by peteshand** [comment]
I've just tested your latest push.. works like a charm!
